src/sentry/issues/occurrence_status_change.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import Sequence
+
+from sentry.models.group import Group, GroupStatus
+from sentry.types.activity import ActivityType
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class OccurrenceStatusChange:
+    fingerprint: Sequence[str]
+    project_id: int
+    new_status: int
+    new_substatus: int | None
+
+    def update_status_for_occurrence(self, group):
+        if group.status == self.new_status and group.substatus == self.new_substatus:
+            return
+
+        # Validate the provided status and substatus - we only allow setting a substatus for unresolved or ignored groups.
+        if self.new_status in [GroupStatus.UNRESOLVED, GroupStatus.IGNORED]:
```

Comment:
so this is a `no-op` for now?

Reply:
yea, the changes to set the statuses are in future PRs